src/simsopt/field/force.py

```diff
@@ -1,118 +1,260 @@
 """Implements the force on a coil in its own magnetic field and the field of other coils."""
-import math
 from scipy import constants
 import numpy as np
 import jax.numpy as jnp
 from jax import grad
 from .biotsavart import BiotSavart
 from .selffield import B_regularized_pure, B_regularized, regularization_circ, regularization_rect
-from ..geo.jit import jit
+from jax import jit
 from .._core.optimizable import Optimizable
 from .._core.derivative import derivative_dec
 
 Biot_savart_prefactor = constants.mu_0 / 4 / np.pi
 
-
 def coil_force_pure(B, I, t):
     """force on coil for optimization"""
-    force = jnp.cross(I * t, B)
-    return force
-
+    return jnp.cross(I * t, B)
 
 def self_force(coil, regularization):
     """
     Compute the self-force of a coil.
     """
     I = coil.current.get_value()
-    tangent = coil.curve.gammadash() / np.linalg.norm(coil.curve.gammadash(), axis=1)[:, None]
+    tangent = coil.curve.gammadash() / np.linalg.norm(coil.curve.gammadash(),axis=1)[:, None]
     B = B_regularized(coil, regularization)
     return coil_force_pure(B, I, tangent)
 
-
 def self_force_circ(coil, a):
     """Compute the Lorentz self-force of a coil with circular cross-section"""
     return self_force(coil, regularization_circ(a))
 
-
 def self_force_rect(coil, a, b):
     """Compute the Lorentz self-force of a coil with rectangular cross-section"""
     return self_force(coil, regularization_rect(a, b))
 
-
 @jit
-def force_opt_pure(gamma, gammadash, gammadashdash,
-                   current, phi, B_ext, regularization):
-    """Cost function for force optimization. Optimize for peak self force on the coil (so far)"""
+def selfforce_opt_pure(gamma, gammadash, gammadashdash, quadpoints, current, regularization):
```

Comment:
The function `selfforce_opt_pure` appears to be unused, so should it be removed?

Reply:
thanks for pointing that out!